src/components/rows/product.js

```diff
@@ -1,91 +1,95 @@
-                            </Row>
-                        </Container>
-                    </ModalBody>
-                    <ModalFooter>
-                        <Button color="success" size="sm" onClick={this.toggle}>Buy Now</Button>{' '}
-                        <Button color="primary" size="sm" onClick={this.toggle}>Cancel</Button>
-                    </ModalFooter>
-                </Modal>
-                <div className="details">
-                    <h6>
-                        {this.props.title}
-                        <br />
-                        <Badge color="secondary" className="badge">{this.props.tag}</Badge>
-                    </h6>
-                    <ButtonToolbar>
-                        <ButtonGroup>
-                            <Button className="button" color="success" size="sm" active>Buy Now</Button>{' '}
-                            <Button className="button" color="light" size="sm" onClick={this.toggle}>More Info</Button>{' '}
-                        </ButtonGroup>
-                    </ButtonToolbar>
-                </div>
-            </div>
-        );
-    }
+  showModal = () => {
+    this.setState({
+      visible: true,
+    });
+  }
+
+  handleOk = () => {
+    this.setState({ loading: true });
+    setTimeout(() => {
+      this.setState({ loading: false, visible: false });
+    }, 5000);
+  }
+
+  handleCancel = () => {
+    this.setState({ visible: false });
+  }
+
+  render() {
+    const { visible, loading } = this.state;
+    return (
+             <div  className={biggestDiv}>
+                <img src={this.props.pix} alt="#" className={product_image} onClick={this.showModal} />
+                <Modal title={this.props.title} centered visible={visible} onOk={this.handleOk} onCancel={this.handleCancel}
+                  footer={[
+                    <Button key="submit" size={small} type={primary} loading={loading} onClick={this.handleOk}>{Buy}</Button>,
+                     <Button key="back" size={small} type={dashed} onClick={this.handleCancel}>{Cancel}</Button>,
+                     ]} >
+
+                    {/* this is the modal content */}
+
+                    <Row type={flex} justify={center}>
+                       <Col className={details}>
+                         <Image src={this.props.pix} thumbnail/>
+                         <br />
+                        <div  className={modalImageDetails}>
+                          <h6><b>Name:</b> {this.props.title}</h6>
+                          <h6><b>Location:</b> {this.props.location}</h6>
+                          <h6><b>Cost:</b> &nbsp; {this.props.tag}</h6>
+                          <h6><b>Quantity available:</b> {this.props.quantity}</h6>
+                          <h6><b>Phone no:</b> {this.props.phone}</h6>
+                          <h6><b>Email:</b> {this.props.email}</h6>
+                       </div>
+                     </Col>
+                   </Row>
+             </Modal>
```

Comment:
Obinna, pull `line 42 - 64` into its own file. This modal should be a sub-component for the Product component. And it will simplify this component here